MonstieWash/Assets/Scripts/Eraser.cs

```diff
@@ -0,0 +1,233 @@
+                newColors[i].b = colors[i].b;
+                newColors[i].a = Mathf.Min((255 - maskPixels[i])/255f, colors[i].a);
+            }
+
+            sprite.texture.SetPixels(newColors, 0);
+            sprite.texture.Apply(false);
+        }
+    }
+
+    private void Awake()
+    {
+        InitializeTool();
+        PopulateErasables();
+    }
+
+    private void Update()
+    {
+        transform.localScale = Vector3.one * tool.size;
+
+        if (!Input.GetMouseButton(0)) return;
+        if (!MouseMoved()) return;
+
+        foreach (var erasable in m_erasables)
+        {
+            if (UpdateErasableMask(erasable)) erasable.ApplyMask();
+        }
+    }
+
+    /// <summary>
+    /// Sets up the tool and related variables ready for use.
+    /// </summary>
+    private void InitializeTool()
+    {
+        tool.Initialize();
+        transform.localScale = Vector3.one * tool.size;
+        GetComponent<SpriteRenderer>().sprite = tool.mask;
+    }
+
+    /// <summary>
+    /// Populates the erasables list, setting up the sprites ready for drawing.
+    /// </summary>
+    private void PopulateErasables()
+    {
+        var tempErasables = GameObject.FindGameObjectsWithTag("Erasable");
+
+        foreach (var erasable in tempErasables)
+        {
+            var spriteRenderer = erasable.GetComponent<SpriteRenderer>();
+            spriteRenderer.sprite = DuplicateSprite(spriteRenderer.sprite);
+            var sprite = spriteRenderer.sprite;
+
+            var newErasable = new Erasable(erasable, sprite, new byte[sprite.texture.width * sprite.texture.height]);
+            m_erasables.Add(newErasable);
+        }
+    }
+
+    /// <summary>
+    /// Duplicates the given sprite.
+    /// </summary>
+    /// <param name="sprite">The sprite to duplicate.</param>
+    /// <returns>The duplicated sprite.</returns>
+    private Sprite DuplicateSprite(Sprite sprite)
+    {
+        var textureCopy = new Texture2D(sprite.texture.width, sprite.texture.height, sprite.texture.format, false);
+        textureCopy.alphaIsTransparency = true;
+        textureCopy.name = $"{sprite.texture.name} Texture Duplicate";
+
+        textureCopy.SetPixels(sprite.texture.GetPixels());
+        textureCopy.Apply();
+
+        var spriteCopy = Sprite.Create(textureCopy, sprite.rect, new Vector2(0.5f, 0.5f), sprite.pixelsPerUnit, 0, SpriteMeshType.FullRect);
+        spriteCopy.name = $"{sprite.name} Sprite Duplicate";
+
+        return spriteCopy;
+    }
+
+    /// <summary>
+    /// Checks if the mouse moved since last frame, updating the last position if it did.
+    /// </summary>
+    /// <returns>Whether or not the mouse moved since last frame.</returns>
+    private bool MouseMoved()
+    {
+        var transformScreenPoint = Camera.main.WorldToScreenPoint(transform.position);
+        m_mouseScreenPos = new Vector2Int(Mathf.RoundToInt(transformScreenPoint.x), Mathf.RoundToInt(transformScreenPoint.y));
```

Comment:
Would be good to have this follow the hand instead of the mouse so we don't have 2 different controls

Reply:
Ah yep most of this was done before your feature was added just for testing, I can update it to include your work. 👍